command/agent/local.go

```diff
@@ -389,16 +389,15 @@ func (l *localState) setSyncState() error {
 		l.serviceStatus[id] = syncStatus{inSync: equal}
 	}
 
+	// Index the remote health checks to improve efficiency
```

Comment:
This change LGTM but in the thread and in the HC conversation he said there are no checks registered. I was expecting an n^2 services reconciliation here but I'm not seeing it.